core/os/android/adb/device.go

```diff
@@ -350,6 +350,15 @@ func scanDevices(ctx context.Context) error {
 	return nil
 }
 
+// rescanDevice force-refresh the device info for a particular device.
+func rescanDevice(ctx context.Context, d bind.Device) error {
+	registry.RemoveDevice(ctx, d)
+	cacheMutex.Lock()
+	delete(cache, d.Instance().Serial)
+	cacheMutex.Unlock()
+	return scanDevices(ctx)
```

Comment:
This seems a bit drastic. Will removing and re-adding a device change its identifier? If so, that will cause all sorts of other issues.

Reply:
By installing or updating ANGLE, we effectively change the device configuration, and hence its ID.

A device instance ID is computed via `core/os/device/instance.go:GenID()`:

```
func (i *Instance) GenID() {
	key := fmt.Sprintf("%v:%v:%v", i.Serial, i.Name, i.Configuration.String())
	i.ID = NewID(id.OfString(key))
}
```

And the configuration contains ANGLE details, from `core/os/device/device.proto`:

```
message Configuration {
...
  // ANGLE info.
  ANGLE angle = 6;
}

...

message ANGLE {
  string package = 1;
  int32 version = 2;
}
```

So in the current situation I believe it is necessary to remove the previous device instance and rescan the device entirely to get a fresh set of information.

The idea behing this change is that whenever we install an APK on a device, we might be changing some parts of the device configuration, so we rescan it, and the effect should be similar to unplugging the device, install the APK on it separately, and plug it in again.

Another possible approach could be to not take the ANGLE version into consideration for computing the ID, such that we can just update ANGLE details while keeping the same ID, but I get the feeling it goes against the current approach of computing the ID as a hash of all configuration details.